node/actors/bft/src/tests/unit_tests.rs

```diff
@@ -0,0 +1,599 @@
+//             Ok(res)
+//         })
+//         .join(ctx)
+//     })
+//         .await
+//         .unwrap();
+//
+//     assert_eq!(
+//         res,
+//         Err(LeaderPrepareError::InvalidPrepareQC(anyhow!("PrepareQC contains messages for different views!")))
+//     );
+// }
+
+mod util {
+    use crate::{
+        io::{InputMessage, OutputMessage},
+        leader::ReplicaPrepareError,
+        replica::LeaderPrepareError,
+        Consensus,
+    };
+    use rand::{rngs::StdRng, Rng, SeedableRng};
+    use zksync_concurrency::{ctx, ctx::Ctx, scope};
+    use zksync_consensus_network::io::ConsensusInputMessage;
+    use zksync_consensus_roles::{
+        validator,
+        validator::{
+            BlockHeader, ConsensusMsg, LeaderPrepare, Payload, Phase, PrepareQC, ReplicaCommit,
+            ReplicaPrepare, SecretKey, Signed, ViewNumber,
+        },
+    };
+    use zksync_consensus_utils::pipe::DispatcherPipe;
+
+    pub(crate) struct Util {
+        ctx: Ctx,
+        consensus: Consensus,
+        rng: StdRng,
+        pipe: DispatcherPipe<InputMessage, OutputMessage>,
+        keys: Vec<SecretKey>,
+    }
+
+    impl Util {
+        pub async fn make() -> Util {
+            Util::make_with(1).await
+        }
+
+        pub async fn make_with(num_validators: i32) -> Util {
+            let ctx = ctx::test_root(&ctx::RealClock);
+            let mut rng = StdRng::seed_from_u64(6516565651);
+            let keys: Vec<_> = (0..num_validators).map(|_| rng.gen()).collect();
+            let (genesis, val_set) =
+                crate::testonly::make_genesis(&keys, validator::Payload(vec![]));
+            let (mut consensus, pipe) =
+                crate::testonly::make_consensus(&ctx, &keys[0], &val_set, &genesis).await;
+
+            consensus.leader.view = ViewNumber(1);
+            consensus.replica.view = ViewNumber(1);
+
+            Util {
+                ctx,
+                consensus,
+                rng,
+                pipe,
+                keys,
+            }
+        }
+
+        pub fn own_key(&mut self) -> &SecretKey {
+            &self.consensus.inner.secret_key
+        }
+
+        pub fn key_at(&mut self, index: usize) -> &SecretKey {
+            &self.keys[index]
+        }
+
+        pub fn rng(&mut self) -> &mut StdRng {
+            &mut self.rng
+        }
+
+        pub fn set_leader_view(&mut self, view: ViewNumber) {
+            self.consensus.leader.view = view
+        }
+
+        pub fn set_leader_phase(&mut self, phase: Phase) {
+            self.consensus.leader.phase = phase
+        }
+
+        pub fn set_replica_view(&mut self, view: ViewNumber) {
+            self.consensus.replica.view = view
+        }
+
+        pub fn set_replica_phase(&mut self, phase: Phase) {
+            self.consensus.replica.phase = phase
+        }
+
+        pub fn make_replica_prepare(
+            &mut self,
+            mutate_callback: Option<impl FnOnce(&mut ReplicaPrepare)>,
```

Comment:
This looks awkward; you could require a mandatory callback (`mutate_callback: impl FnOnce(_)`) and provide a no-op (`make_replica_prepare(|_| {})`) when it's not needed.

Reply:
Yes it turned out to be clumsy especially because type inference for `None` doesn't work. 
Removed.